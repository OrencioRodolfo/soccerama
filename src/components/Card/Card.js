import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Header from './Header/Header'
import Content from './Content/Content'

const Container = styled.div`
  box-shadow: 2px 2px 18px -4px rgba(0,0,0,0.75);
  border-radius: 5px;
  margin: 10px;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
`

const Card = ({ children }) => (
  <Container>
    {children}
  </Container>
)

Card.Header = Header
Card.Content = Content

Card.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Card
