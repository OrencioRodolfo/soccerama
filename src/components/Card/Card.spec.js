import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card component', () => {
  it('Must render properly', () => {
    const component = shallow(<Card>Some content</Card>)
    expect(component).toMatchSnapshot()
  })

  it('Must export the Header and Content components', () => {
    expect(Card.Header).toBeDefined()
    expect(Card.Content).toBeDefined()
  })
})
