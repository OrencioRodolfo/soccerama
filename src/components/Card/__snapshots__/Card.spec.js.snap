// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    Some content
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Some content",
    },
    "ref": null,
    "rendered": "Some content",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Some content",
      },
      "ref": null,
      "rendered": "Some content",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
